specs/ceph_installer_wrapper.adoc

```diff
@@ -37,103 +37,330 @@ A wrapper needs to be created which abstracts the capabilities provided by the c
 
-=== Developer impact:
+=== Tendrl/node_agent impact:
 
-None.
+Since ceph installer does not support operating from a node inside the cluster,
+the node agent running on the API node needs to be designated as a provisioner
+for ceph cluster. This would be accomplished by adding a tag for
+`provisioners/ceph`. The API would route jobs directly to this tag, with an
+empty `node_ids` array in the job payload.
 
+[source,json]
+.Job structure with all the possible parameters
+----
+{
+  "integration_id": "9a4b84e0-17b3-4543-af9f-e42000c52bfc",
+  "run": "tendrl.flows.CreateCluster",
+  "status": "new",
+  "type": "node",
+  "node_ids": [],
+  "tags": ["provisioner/ceph"],
+  "parameters": {
+    "sds_type": "ceph",
+    "sds_version": "10.2.5",
+    "name": "MyCluster",
+    "TendrlContext.integration_id": "9a4b84e0-17b3-4543-af9f-e42000c52bfc",
+    "Node[]": [
+      "3a95fd96-876d-439a-a64d-70332c069aaa",
+      "3943fab1-9ed2-4eb6-8121-5a69499c4568",
+      "b10e00e9-e444-41c2-9517-df2118b42731"
+    ],
+    "node_configuration": {
+      "3a95fd96-876d-439a-a64d-70332c069aaa": {
+        "role": "ceph/mon",
+        "provisioning_ip": "10.0.0.24"
+      },
+      "3943fab1-9ed2-4eb6-8121-5a69499c4568": {
+        "role": "ceph/osd",
+        "provisioning_ip": "10.0.0.29",
+        "journal_size": 5192,  # Megabytes
+        "journal_colocation": "false",
+        "storage_disks": [
+          {
+            "device": "/dev/sda",
+            "journal": "/dev/sdc"
+          },
+          {
+            "dev_name": "/dev/sdb",
+            "journal": "/dev/sdc"
+          }
+        ]
+      },
+      "b10e00e9-e444-41c2-9517-df2118b42731": {
+        "role": "ceph/osd",
+        "provisioning_ip": "10.0.0.30",
+        "journal_colocation": "true",
+        "storage_disks": [
+          {
+            "device": "/dev/sda"
+          },
+          {
+            "dev_name": "/dev/sdb"
+          }
+        ]
+      }
+    }
+  }
+}
+----
+
+The validations to be carried out on these parameters are as follows, when the
+role of a node is `ceph/osd`:
+
+* `journal_size` is optional. Value should be in Megabytes.
+* `journal_colocation` is optional and value should be a boolean.
+* `storage_disks` is mandatory. If not provided, it's an error because the node
+  cannot be part of the cluster as an osd node.
+** It has to be an array.
+** It has to contain hashes.
+** Each hash has to contain a `device` key.
+*** If `journal_colocation` is set to `true` _for the node_, the `journal` key
+    cannot exist in the hash
+*** If `journal_colocation` is set to `false` _for the node_, the `journal` key
+    must contain a device
+** Every device path specified in either the `device` or the `journal` key must
+   be a valid block device on the node
+** A device cannot be listed under both `device` and `journal` (across multiple
+   hashes), per host
```

Comment:
Single device should not be listed under different hashes, eg. this seems wrong to me and it would be easy to verify:

```
"storage_disks": [
  {
    "device": "/dev/sda"
  },
  {
    "device": "/dev/sda"
  }
```